terraform/main.tf

```diff
@@ -0,0 +1,234 @@
+  vpc_id = "${aws_vpc.data_refinery_vpc.id}"
+  # Unsure if this should be set to true
+  map_public_ip_on_launch = true
+
+  tags {
+    Name = "data-refinery-1b"
+  }
+}
+
+resource "aws_iam_instance_profile" "ecs_instance_profile" {
+  name  = "data-refinery-ecs-instance-profile"
+  roles = ["${aws_iam_role.ecs_instance.name}"]
+}
+
+resource "aws_iam_role" "ecs_instance" {
+  name = "data-refinery-ecs-instance"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "s3_access_policy" {
+  name = "data-refinery-s3-access-policy"
+  description = "Allows S3 Permissions."
+
+  policy = <<EOF
+{
+   "Version":"2012-10-17",
+   "Statement":[
+      {
+         "Effect":"Allow",
+         "Action":[
+            "s3:ListAllMyBuckets"
+         ],
+         "Resource":"arn:aws:s3:::*"
+      },
+      {
+         "Effect":"Allow",
+         "Action":[
+            "s3:ListBucket",
+            "s3:GetBucketLocation"
+         ],
+         "Resource":"arn:aws:s3:::data-refinery"
+      },
+      {
+         "Effect":"Allow",
+         "Action":[
+            "s3:PutObject",
+            "s3:GetObject",
+            "s3:DeleteObject"
+         ],
+         "Resource":"arn:aws:s3:::data-refinery/*"
+      }
+   ]
+}
+EOF
+}
+
+
+resource "aws_iam_policy_attachment" "ecs" {
+  name = "AmazonEC2ContainerServiceforEC2Role"
+  roles = ["${aws_iam_role.ecs_instance.name}"]
+
+  # The following can be found here:
+  # https://console.aws.amazon.com/iam/home?region=us-east-1#/policies/arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
+}
+
+resource "aws_iam_policy_attachment" "s3" {
+  name = "data-refinery-s3-access-policy-attachment"
+  roles = ["${aws_iam_role.ecs_instance.name}"]
+
+  policy_arn = "${aws_iam_policy.s3_access_policy.arn}"
+}
+
+resource "aws_key_pair" "data_refinery" {
+  key_name = "data-refinery-key"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDG1WFcvLLyLK7jZfhYnNBfbVwm259du2ig4tYcicA1d8d8I43LcWg2WYpd7EfNFH8LnJMDg632NcnQ0qzrpUG4zLGTcYufXm1Fm97J285iabzlUxfgSpbk5Ee1ioNCmqtPxEgy5lrt2xw0p3Rnbn0NvSKzwGU82/k/NCbxeKbaRpHLjz9TTcAdcZLugV7Syr8W+zWBqlCIMyC4ce4t8s/ecGbyacmRPdPqC9jUBC0guLHeQmlinINJIr+wMihxJ0B5Zcyokf4wXlQBPPcB89oO9L81nlApY6aK5JJrhkSN8M5+YOkdk6Xi4SZuJD5SLWbilKGPiCNiLPAnPw7m7Ual"
+}
+
+
+
+resource "aws_internet_gateway" "data_refinery" {
+  vpc_id = "${aws_vpc.data_refinery_vpc.id}"
+
+  tags = {
+    Name = "data-refinery"
+  }
+}
+
+# Note: this is a insecure practice long term, however it's
+# necessary to access it from lab machines.
+resource "aws_route_table" "data_refinery" {
+  vpc_id = "${aws_vpc.data_refinery_vpc.id}"
+
+  route {
+    cidr_block = "0.0.0.0/0"
```

Comment:
Yes. It's the way to denote all IPv4 traffic. (Since the /0 part means that none of the IP address is significant.)